### PR TITLE
Add information about overriding Tickless mode.

### PR DESCRIPTION
**Pull request by hugueskamba**
Add a section to the end of the power management handbook page to disable tickless mode from app config json file.

### CODE REVIEW

File: docs/api/platform/PowerManagement.md [hunk 1]
```diff
@@ -81,6 +81,26 @@ UNLOCK: mbed_rtx_idle.cpp, ln: 131, lock count: 1
 
 <span class="notes">**Note:** Sleep tracing is a debug feature and should only be enabled during development cycle. Its heavy use of UART can affect the device performance.</span>
 
+By default, tickless mode is enabled for some targets that support it.
+
+You can override the default configuration in your application `mbed_app.json` and achieve further memory optimisation if tickless mode does not fit your application requirement as you may rely on SysTick always being present; then .
```
Comment (evedon) [inline docs/api/platform/PowerManagement.md]: I would add a sentence at the start to explain that there is a trade off between power saving and memory footprint. Something like:
"There is a small trade-off between power saving and memory footprint. When tickless mode is enabled, the memory footprint of your application is slightly bigger. Therefore if tickless mode does not fit your application requirement, for example you may rely on SysTick always being present, then you can disable tickless and achieve further memory optimisation. To do that, you can override the default configuration in your application `mbed_app.json` like this:"

Comment (hugueskamba) [reply docs/api/platform/PowerManagement.md]: [Done](https://github.com/ARMmbed/mbed-os-5-docs/compare/d406b844efbabb14ebdb353bdb395bad023c4139..e6393fa5d1a2e54e35f14d29ab4cf25809a7f102).

**Review by SeppoTakalo — APPROVED** [review]

File: docs/api/platform/PowerManagement.md [hunk 2]
```diff
@@ -81,6 +81,22 @@ UNLOCK: mbed_rtx_idle.cpp, ln: 131, lock count: 1
...
 
+There is a small trade-off between power saving and memory footprint. When tickless mode is enabled, the memory footprint of your application is slightly bigger. Therefore if tickless mode does not fit your application requirement, for example you may rely on SysTick always being present, then you can disable tickless and achieve further memory optimisation. To do that, you can override the default configuration in your application `mbed_app.json` like this:
+
+```json
+{
+    "macros_remove": ["MBED_TICKLESS"]
+}
+```
+
+https://github.com/ARMmbed/mbed-os-example-blinky application compiled with the GCC_ARM toolchain for some targets.
```
Comment (bulislaw) [inline docs/api/platform/PowerManagement.md]: I would remove these information as it is bound to go out of date. I'd rather quote the range of difference eg. between 500B and 1k

Comment (hugueskamba) [reply docs/api/platform/PowerManagement.md]: [Done](https://github.com/ARMmbed/mbed-os-5-docs/compare/2015e28ce4d2149a6760b34709b4fb4386329b52..e46ca9f89262cd51257d899408d1442c694c0b6a).

File: docs/api/platform/PowerManagement.md [hunk 3]
```diff
@@ -81,6 +81,16 @@ UNLOCK: mbed_rtx_idle.cpp, ln: 131, lock count: 1
 
 <span class="notes">**Note:** Sleep tracing is a debug feature and should only be enabled during development cycle. Its heavy use of UART can affect the device performance.</span>
 
+There is a small trade-off between power saving and memory footprint. When tickless mode is enabled, the memory footprint of your application is slightly bigger. Therefore, if tickless mode does not fit your application requirement, you can disable tickless and achieve further memory optimization. To do that, override the default configuration in the application `mbed_app.json`:
```
Comment (AnotherButler) [inline docs/api/platform/PowerManagement.md:84]: Would this content fit better in our power optimization section (https://os.mbed.com/docs/mbed-os/v5.14/tutorials/power-optimization.html)?

Comment (hugueskamba) [reply docs/api/platform/PowerManagement.md:84]: 🤷... @bulislaw ?

Comment (bulislaw) [reply docs/api/platform/PowerManagement.md:84]: No, the power management page should be referenced in the tutorial. What we are focusing on here is "this low power feature which is on by default may not always work for you".

**Review by AnotherButler — APPROVED** [review]

**Comment by AnotherButler** [comment]
Is this content ready to be made live, or is there any sort of code dependency?

**Comment by hugueskamba** [comment]
> Is this content ready to be made live, or is there any sort of code dependency?

There are no dependencies.

**Comment by hugueskamba** [comment]
This [force-push](https://github.com/ARMmbed/mbed-os-5-docs/compare/86d70d49c824c75c3ca61ed7564ea48bc9c40dd4..31b955120422638825396eb2422a51cd4881c830) clarifies how to remove the macro.

**Pull request merged**